conans/search/search.py

```diff
@@ -99,16 +99,12 @@ def search_recipes(cache, pattern=None, ignorecase=True):
         pattern = re.compile(pattern, re.IGNORECASE) if ignorecase else re.compile(pattern)
 
     subdirs = list_folder_subdirs(basedir=cache.store, level=4)
-    if not pattern:
-        return sorted([ConanFileReference(*folder.split("/")) for folder in subdirs])
-    else:
-        ret = []
-        for subdir in subdirs:
-            ref = ConanFileReference(*subdir.split("/"))
-            if _partial_match(pattern, ref):
-                ret.append(ref)
-
-        return sorted(ret)
+    refs = [ConanFileReference(*folder.split("/")) for folder in subdirs]
+    refs.extend(cache.editable_packages.refs().keys())
+    refs = sorted(refs)
```

Comment:
Order the list after applying the filter (scratch a little performance)

Reply:
True, done